prow/pod-utils/wrapper/options.go

```diff
@@ -78,49 +83,71 @@ func (o *Options) Validate() error {
 	return nil
 }
 
-func WaitForMarker(ctx context.Context, path string) (int, error) {
+func WaitForMarkers(ctx context.Context, paths ...string) map[string]MarkerResult {
 	// Only start watching file events if the file doesn't exist
 	// If the file exists, it means the main process already completed.
-	if _, err := os.Stat(path); os.IsNotExist(err) {
-		watcher, err := fsnotify.NewWatcher()
-		if err != nil {
-			return -1, fmt.Errorf("new fsnotify watch: %v", err)
-		}
-		defer watcher.Close()
-		dir := filepath.Dir(path)
-		if err := watcher.Add(dir); err != nil {
-			return -1, fmt.Errorf("add %s to fsnotify watch: %v", dir, err)
+
+	results := make(map[string]MarkerResult)
+
+	if len(paths) == 0 {
+		return results
+	}
+
+	for _, path := range paths {
+		if _, err := os.Stat(path); !os.IsNotExist(err) {
+			results[path] = readMarkerFile(path)
 		}
+	}
+	watcher, err := fsnotify.NewWatcher()
+	if err != nil {
+		return results
```

Comment:
I think this overall needs better error handling -- callsites expect the returned map to have a meaningful `MarkerResult` for each path they passed in, and returning it possibly empty means they can receive a `MarkerResults` zero value, which is `ReturnCode==0 Err=nil`, which is the exact opposite of the truth.

So IMO we should either populate the map with default error values (meaning something like "did not even attempt to check"), or this method should have a `error` of its own, meaning "all paths have meaningful `MarkerResults`" when not nil.

Reply:
Agreed, thank you for pointing this out.  I think I am inclined to implement the "populate the map with default error values" approach; that seems like the more intuitive behavior IMO.  I'll go ahead and make this change.